rotkehlchen/history/price.py

```diff
@@ -155,38 +186,50 @@ def get_price_for_special_asset(
             if to_asset == A_USD:
                 return usd_price
 
-            price_mapping = PriceHistorian().query_historical_price(
+            usd_to_target_price = PriceHistorian._get_cached_price_or_query(
                 from_asset=A_USD,
                 to_asset=to_asset,
                 timestamp=timestamp,
+                max_seconds_distance=max_seconds_distance,
             )
-            return Price(usd_price * price_mapping)
+            return Price(usd_price * usd_to_target_price) if usd_to_target_price is not None else None  # noqa: E501
 
         if from_asset == A_POLYGON_POS_MATIC and timestamp > POLYGON_POS_POL_HARDFORK:
-            return PriceHistorian.query_historical_price(
-                from_asset=Asset('eip155:1/erc20:0x455e53CBB86018Ac2B8092FdCd39d8444aFFC3F6'),  # POL token  # noqa: E501,
+            return PriceHistorian._get_cached_price_or_query(
+                from_asset=Asset('eip155:1/erc20:0x455e53CBB86018Ac2B8092FdCd39d8444aFFC3F6'),  # POL token  # noqa: E501
```

Comment:
Do a grep for this and wherever it is used replace it with a constant asset from the asset constants. The reason for this is that if they ever delete it from the DB this will fail very badly. This is why for hard coded stuff we have the constant assets.